src/usePreviousDistinct/usePreviousDistinct.ts

```diff
@@ -0,0 +1,31 @@
+import { useRef } from 'react';
+import { useFirstMountState } from '..';
+
+export type Predicate<T> = (prev: T | undefined, next: T) => boolean;
+
+const checkIfStrictlyEqual = <T>(prev: T | undefined, next: T) => prev === next;
+
+/**
+ * Returns the most recent _distinct_ value passed to the hook on previous render. Distinct here
+ * means that the hook's return value will only update when the passed value updates. This is
+ * useful when other renders are involved potentially making multiple, irrelavant updates.
+ *
+ * Yields `undefined` on first render.
+ *
+ * @param value Value to yield on next render
+ */
+export function usePreviousDistinct<T>(
+  value: T,
+  compare: Predicate<T> = checkIfStrictlyEqual
```

Comment:
I think `comparator` or `predicate` is better.
Also this argument is not listed in JSDoc